### PR TITLE
Implement value_and_grad option.

### PR DESCRIPTION
**Pull request by mblondel**

### CODE REVIEW

File: jaxopt/_src/gradient_descent.py [hunk 1]
```diff
@@ -33,19 +33,28 @@ class GradientDescent(ProximalGradient):
     fun: a smooth function of the form ``fun(parameters, *args, **kwargs)``,
       where ``parameters`` are the model parameters w.r.t. which we minimize
       the function and the rest are fixed auxiliary parameters.
+    value_and_grad: whether ``fun`` just returns the value (False) or both
+      the value and gradient (True).
+    has_aux: whether ``fun`` outputs auxiliary data or not.
```
Comment (fllinares) [inline jaxopt/_src/gradient_descent.py:38]: Probably obvious from the context, but the documentation for `has_aux` only describes the output signature of `fun` when `has_aux == True`. Should we explain `has_aux == False` too?

**Review by fllinares — APPROVED** [review]
Thanks a lot Mathieu, looks great! Just left a minor question regarding the doc (which applies to all solvers).

**Review by froystig — APPROVED** [review]
Looks good!

**Comment by mblondel** [comment]
Thanks both for the reviews! I addressed Felipe's comment.

**Pull request merged**